packages/flutter/lib/src/services/predictive_back_event.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+/// Enum representing the edge from which a swipe starts in a back gesture.
+///
+/// This is used in [PredictiveBackEvent] to indicate the starting edge of the
+/// swipe gesture.
+enum SwipeEdge {
+  /// Indicates that the swipe gesture starts from the left edge of the screen.
+  left,
+
+  /// Indicates that the swipe gesture starts from the right edge of the screen.
+  right,
+}
+
+/// Object used to report back gesture progress in Android.
+///
+/// Holds information about the touch event, swipe direction and the animation
+/// progress that predictive back animations should seek to.
+@immutable
+class PredictiveBackEvent {
+  /// Creates a new [PredictiveBackEvent] instance.
+  const PredictiveBackEvent._({
+    required this.x,
+    required this.y,
+    required this.progress,
+    required this.swipeEdge,
+  }) : assert(progress >= 0.0 && progress <= 1.0);
+
+  /// Creates an [PredictiveBackEvent] from a Map, typically used when converting
+  /// data received from a platform channel.
+  factory PredictiveBackEvent.fromJSON(Map<String, dynamic> json) {
+    return PredictiveBackEvent._(
+      x: (json['x'] as num?)?.toDouble(),
+      y: (json['y'] as num?)?.toDouble(),
+      progress: (json['progress'] as num).toDouble(),
+      swipeEdge: SwipeEdge.values[json['swipeEdge'] as int],
+    );
+  }
+
+  /// The global X location of the touch point, or `null` if the event is from a
+  /// button press.
+  final double? x;
+
+  /// The global Y location of the touch point, or `null` if the event is from a
+  /// button press.
+  final double? y;
+
+  /// Returns a value between 0 and 1 representing how far along the back
+  /// gesture is.
+  ///
+  /// This value is driven by the horizontal location of the touch point, and
+  /// should be used as the fraction to seek the predictive back animation with.
+  /// Specifically,
+  ///
+  /// - The progress is 0 when the touch is at the starting edge of the screen
+  ///   (left or right), and the animation should seek to its start state.
+  /// - The progress is approximately 1 when the touch is at the opposite side
+  ///   of the screen, and the animation should seek to its end state. Exact end
+  ///   value may vary depending on screen size.
+  ///
+  /// When the gesture is canceled, the progress value continues to update,
+  /// animating back to 0 until the cancellation animation completes.
+  ///
+  /// In-between locations are linearly interpolated based on horizontal
+  /// distance from the starting edge and smooth clamped to 1 when the distance
+  /// exceeds a system-wide threshold.
+  final double progress;
+
+  /// The screen edge from which the swipe gesture starts.
+  final SwipeEdge swipeEdge;
```

Comment:
Ah you're right. The default predictive back animation does not use this (I double checked native just now, swiping from the right does the exact same animation as swiping from the left), however, OpenContainer animations do use this. I've tried to keep this PR as minimal as possible, but I advocate for keeping swipeEdge, since I'm already using it right now as I try to implement OpenContainer animations in the `animations` package.

Reply:
If I have a 2-panel application with 2 navigators side-by-side, do I use `backEvent.position` / `backEvent.swipeEdge` to determine which navigator should respond to the back gesture? But for button events which attribute should I use, is `swipeEdge` relevant when it's a button event?